src/ios-deploy/ios-deploy.m

```diff
@@ -1567,6 +1589,73 @@ void remove_path(AMDeviceRef device) {
     check_error(AFCConnectionClose(afc_conn_p));
 }
 
+void remove_tree(afc_connection* afc_conn_p, const char* dir)
+{
+    char *dir_ent;
+    
+    afc_dictionary* afc_dict_p;
+    char *key, *val;
+    int not_dir = 0;
+    
+    unsigned int code = AFCFileInfoOpen(afc_conn_p, dir, &afc_dict_p);
+    if (code != 0) {
+        // there was a problem reading or opening the file to get info on it, abort
+        return;
+    }
+    
+    while((AFCKeyValueRead(afc_dict_p,&key,&val) == 0) && key && val) {
+        if (strcmp(key,"st_ifmt")==0) {
+            not_dir = strcmp(val,"S_IFDIR");
+        }
+    }
+    AFCKeyValueClose(afc_dict_p);
+    
+    if (not_dir) {
+        // Not a directory; asssume it's a leaf file - delete it
+        NSLogVerbose(@"Deleting file %s", dir);
+        assert(AFCRemovePath(afc_conn_p, dir) == 0);
+        return;
+    }
```

Comment:
A lot of this seems to be duplicated from read_dir(). Is there a reason this can't be done with read_dir() and a callback like download_tree() does?